tests/nsrt_learning/test_sampler_learning.py

```diff
@@ -111,3 +113,44 @@ def test_learn_samplers_failure():
                                  lambda s, m, o, p: False)
     with pytest.raises(NotImplementedError):  # bad sampler_learner
         learn_samplers([None], None, [(option, [])], "bad sampler learner")
+
+
+def test_learned_sampler_with_goal():
+    """Tests _LearnedSampler() when goals are used."""
+    utils.reset_config({
+        "env": "cluttered_table_place",
+        "sampler_learning_use_goals": True,
+        "num_train_tasks": 15,
+        "neural_gaus_regressor_max_itr": 3
+    })
+    env = ClutteredTablePlaceEnv()
+    task = env.get_test_tasks()[0]
+    goal = task.goal
+
+    rng = np.random.default_rng(0)
+    state = task.init
+    goal = task.goal
+    goal_obj = next(iter(goal)).objects[0]
+    objects = [goal_obj]
+
+    input_size = 11
+    classifier = MLPClassifier(input_size, 10000)
```

Comment:
How is the speed of this test? 10000 looks scarily large.

Reply:
`pytest tests/nsrt_learning/test_sampler_learning.py --durations=0`

Reply:
The classifier isn't used for anything other than to initialize the _LearnedSampler, so the speed of the test isn't really affected (~2.3s). Should I change it to be smaller anyway?

Reply:
do you mean the speed of just this one function you added, or the entire `test_sampler_learning.py`?

Reply:
Hmm sorry, 2.3s is the speed of the entire file. Seems like without the new function the file takes ~2.1s to run.

Reply:
okay not a problem